src/System.Text.JsonLab.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -20,6 +21,78 @@ private JsonDynamicObject(Dictionary<JsonProperty, JsonValue> properties)
             _properties = properties;
         }
 
+        private static Dictionary<Type, Dictionary<int, PropertyInfoLinkedList>> TypeCache = new Dictionary<Type, Dictionary<int, PropertyInfoLinkedList>>();
+
+        public static T Deserialize<T>(ReadOnlySpan<byte> utf8)
+        {
+            if (!TypeCache.TryGetValue(typeof(T), out Dictionary<int, PropertyInfoLinkedList> dictionary))
+            {
+                dictionary = GetTypeMap(typeof(T));
+                TypeCache.Add(typeof(T), dictionary);
+            }
```

Comment:
@Maoni0, this is a cache where we might want to experiment with the memory APIs.